File: backend/src/main/java/com/kadimar/management/model/Document.java
package com.kadimar.management.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "documents")
public class Document {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String type;

    @Column(nullable = false)
    private String path;

    private String description;

    @Column(nullable = false)
    private LocalDateTime uploadedAt;

    @ManyToOne
    @JoinColumn(name = "uploaded_by")
    private User uploadedBy;

    @ManyToOne
    @JoinColumn(name = "project_id")
    private Project project;

    @Column(nullable = false)
    private Long size;

    @Column(nullable = false)
    private String contentType;

    @PrePersist
    protected void onCreate() {
        uploadedAt = LocalDateTime.now();
    }
}

File: types/document.ts
export interface Document {
  id: string;
  name: string;
  type: string;
  description?: string;
  uploadedAt: string;
  uploadedBy: string;
  projectId: string;
  size: number;
  contentType: string;
}

export interface DocumentUpload {
  file: File;
  projectId: string;
  description?: string;
}